LibreHardwareMonitor/Utilities/HttpServer.cs

```diff
@@ -563,16 +569,106 @@ private async Task SendJsonAsync(HttpListenerResponse response, HttpListenerRequ
+            {
+                string prometheusHost = node.Parent.Text;
+                foreach (SensorNode sensor in node.Nodes[i].Nodes)
+                {
+                    string tagHardware = (((HardwareNode)node).Hardware.Parent != null ? ((HardwareNode)node).Hardware.Parent.HardwareType.ToString() : ((HardwareNode)node).Hardware.HardwareType.ToString());
+                    string tagSensorType = sensor.Sensor.SensorType.ToString();
+                    string tagSensorUnits = (units[sensor.Sensor.SensorType].Item1.Length == 0 ? "" : "_" + units[sensor.Sensor.SensorType].Item1);
+                    string tagName = $"lhm_{tagHardware}_{tagSensorType}{tagSensorUnits}";
+                    tagName = tagName.ToLower();
+
+                    string valueSensor = sensor.Text.Replace("#", "");
+                    string valueHardware = node.Text;
+                    string valueId = sensor.Sensor.Identifier.ToString().Split('/').Last();
+                    string valueFamily = node.Nodes[i].Text;
+                    string valueHost = _root.Text;
+
+                    try
+                    {
+
+                        if (lastTagName != tagName)
+                        {
+                            responseStr += $"# TYPE {tagName} gauge\n";
+                            lastTagName = tagName;
+                        }
+
+                        double? tagValue = units[sensor.Sensor.SensorType].Item2 * sensor.Sensor.Value;
+                        responseStr += $$"""{{tagName}} {"sensor"="{{valueSensor}}" "hardware"="{{valueHardware}}" "id"="{{valueId}}" "family"="{{valueFamily}}" "host"="{{valueHost}}"} {{tagValue}}""" + "\n";
+                    }
+                    catch (Exception)
+                    {
+                        responseStr += $"# HELP {lastTagName} This Sensor type is not defined in the prometheus adapter [{sensor.Sensor.SensorType}]\n";
+                        responseStr += $$"""{{tagName}} {"sensor"="{{valueSensor}}" "hardware"="{{valueHardware}}" "id"="{{valueId}}" "family"="{{valueFamily}}" "host"="{{valueHost}}"} {{sensor.Sensor.Value}}""" + "\n";
+                    }
+                }
+            }
+        }
+        return responseStr;
+    }
+
+    private void SendPrometheus(HttpListenerResponse response, HttpListenerRequest request = null)
+    {
+        string responseContent = SendPrometheusTreeTransversal(_root);
+        SendResponse(response, "text/plain", responseContent);
+    }
+
     private async Task SendJsonSensorAsync(HttpListenerResponse response, Dictionary<string, object> sensorData)
     {
         // Convert the JObject to a JSON string
         string responseContent = System.Text.Json.JsonSerializer.Serialize(sensorData);
+        SendResponse(response, "application/json", responseContent);
+    }
+
+    private async void SendResponse(HttpListenerResponse response, string contentType, string responseContent)
```

Comment:
```suggestion
    private async Task SendResponseAsync(HttpListenerResponse response, string contentType, string responseContent)
```

Reply:
There is a static SendResponseAsync already, I will use that one instead.